### PR TITLE
Update README.md to include Kotlin Examples

### PR DESCRIPTION
**Pull request by sangeetds**
Added Kotlin Examples, as discussed in Issue #1345

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,21 +340,72 @@ class EventJsonAdapter {
...
+
+<details>
+    <summary>Kotlin</summary>
+
+```kotlin
+class EventJsonAdapter {
+  @FromJson fun eventFromJson(eventJson: EventJson): Event {
+    val event = Event()
+    event.title = eventJson.title
+    event.beginDateAndTime = "${eventJson.begin_date}  ${eventJson.begin_time}"
```
Comment (saket) [inline README.md:353]:
```suggestion
    event.beginDateAndTime = "${eventJson.begin_date} ${eventJson.begin_time}"
```

**Review by saket — APPROVED** [review]
LGTM!

@ZacSweers @swankjesse Thoughts on replacing all java blocks with kotlin instead of keeping them both?

File: README.md [hunk 2]
```diff
@@ -341,9 +578,23 @@ class Player {
   ...
 }
 ```
+</details>
 
-Because JSON field names are always defined with their Java fields, Moshi makes it easy to find
-fields when navigating between Java and JSON.
+<details>
```
[the remark this answers is not in the document]
Comment (sangeetds) [reply README.md:585]: You're right. Should I include a message saying it is assuming the use of `KotlinJsonAdapterFactory` and code-gen?

File: README.md [hunk 3]
```diff
@@ -787,11 +787,11 @@ public final class BlackjackHand {
     <summary>Kotlin</summary>
 
 ```kotlin
-class BlackjackHand {
-  @Transient private val total: Int
+class BlackjackHand(
```
Comment (ZacSweers) [inline README.md:790]: This case should have a default value assigned to it or be converted to a non-constructor var property. Probably something like this

```kotlin
class BlackjackHand(...) {
  @Transient
  private var total: Int = 0
}
```

**Review by ZacSweers — APPROVED** [review]
One more nit!

File: README.md (hunk 3 above)
Comment (sangeetds) [reply README.md]: Done.

**Review by ZacSweers — APPROVED** [review]

**Review by saket — APPROVED** [review]

**Comment by sangeetds** [comment]
@ZacSweers can you run the tests again? Thanks

**Comment by ZacSweers** [comment]
Merged. Thanks a ton!

**Pull request merged**